nemo/collections/asr/models/rnnt_spk_bpe_models.py

```diff
@@ -232,17 +422,69 @@ def training_step(self, batch, batch_nb):
     def validation_pass(self, batch, batch_idx, dataloader_idx=0):
```

Comment:
In my understanding, both `training_step` and `validation_pass` only take the results from `forward` and `decoder` function and do loss/wer calculation. All of the model calculation should be done in `forward` function so that you don't need to have duplicate code in `training_step` and `validation_pass`. I am not sure if it is already there when you got the code from our branch. But if you take a look at the `EncDecMultiTaskModel` in `nemo/collections/asr/models/aed_multitask_models.py`, actually you only need to change `forward` and keep the `training_step` and `validation_pass` unchanged. 

That would also make this code a lot simple, otherwise you have to change the code twice every time you have some new ideas. 

In my suggestion, you can have a new `forward_asr` function, which is exactly doing the same thing as the `forward` function does in `rnnt_model`. And you already have a `forward_diar` function. Then you can put everything like concatenation and projection in the `training_step` to the `forward` function and return `encoded`. In this way you don't need to change the `training_step` and `validation_pass` function in the original `rnnt_model` class.